### PR TITLE
Documentation updates for building the docs.

### PR DESCRIPTION
**Pull request by rowanc1**
I had to remind myself how to use sphinx, I have been working in JS land for far too long apparently.

Thought this might be helpful for pointers to other people onboarding from that world.

I also updated a few of the pointers and `ThebeLab` --> `Thebe`

### CODE REVIEW

**Comment by rowanc1** [comment]
Ok, I have merged in the changes from #261.

Looks like:
image

I opted for a h2 header, so it doesn't show up in the main TOC tree.

Also my first time using the cross-ref in myst. :)

File: docs/contribute.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ See the [ExecutableBooks developer guidelines](https://executablebooks.org/en/la
 - `src/` contains the code and assets that make up `thebe`. This is what you'll edit to make changes to the project.
 - `examples/` provides a few HTML examples of how `thebe` can be used. It is mostly for documentation
 
+(dev-install)=
```
Comment (moorepants) [inline docs/contribute.md:20]: Not sure what this does.

Comment (moorepants) [reply docs/contribute.md:20]: I see now below it is some way to make a reference. Never seen this markdown before.

**Comment by moorepants** [comment]
LGTM.

File: docs/contribute.md (hunk 1 above)
Comment (rowanc1) [reply docs/contribute.md:20]: It is MyST!!

**Pull request merged**